pkg/registry/core/service/storage/rest_test.go

```diff
@@ -1451,6 +1451,70 @@ func TestServiceRegistryExternalTrafficGlobal(t *testing.T) {
 	}
 }
 
+// Validate the internalTrafficPolicy field when set to "Cluster" then updated to "Local"
+func TestServiceRegistryInternalTrafficPolicyClusterThenLocal(t *testing.T) {
```

Comment:
I'd prefer these as two table-entries in a single test which also covered the gate being disabled, but I won't hold your feet to that fire this week.

Reply:
These tests are not really testing much, are they?  We should probably just have a single test that permutes all of the "dumb data" (no side-effects) fields, creates them, verifies, updates them, verifies.  Later.

Reply:
ack

Reply:
@hanlins @XudongLiuHarold @nicolehanjing in case any of you are interested in refactoring these tests in a follow-up :)

Reply:
Also, I am working on de-layering service REST (https://github.com/kubernetes/kubernetes/pull/96684) which includes auditing and overhauling tests.